FlagsmithClient/Classes/Flagsmith.swift

```diff
@@ -83,36 +83,52 @@ public final class Flagsmith: @unchecked Sendable {
     ///   - identity: ID of the user (optional)
     ///   - completion: Closure with Result which contains array of Flag objects in case of success or Error in case of failure
     public func getFeatureFlags(forIdentity identity: String? = nil,
+                                traits: [Trait]? = nil,
                                 completion: @Sendable @escaping (Result<[Flag], any Error>) -> Void)
     {
         if let identity = identity {
-            getIdentity(identity) { result in
-                switch result {
-                case let .success(thisIdentity):
-                    completion(.success(thisIdentity.flags))
-                case let .failure(error):
-                    if self.defaultFlags.isEmpty {
-                        completion(.failure(error))
-                    } else {
-                        completion(.success(self.defaultFlags))
+            if let traits = traits {
+                apiManager.request(.postTraits(identity: identity, traits: traits)) { (result: Result<Traits, Error>) in
+                    switch result {
+                    case let .success(result):
+                        completion(.success(result.flags))
+                    case let .failure(error):
+                        self.handleFlagsError(error, completion: completion)
+                    }
+                }
+            } else {
+                getIdentity(identity) { result in
+                    switch result {
+                    case let .success(thisIdentity):
+                        completion(.success(thisIdentity.flags))
+                    case let .failure(error):
+                        self.handleFlagsError(error, completion: completion)
```

Comment:
I think (perhaps in the future) these should be consolidated. There's not really any reason to use the `GET` endpoint here, we might as well just always use the `POST` endpoint, and just exclude the traits if not provided. 

If this is trivial to do, perhaps we can implement this now, but otherwise, we can add it to the backlog for now.

Reply:
#65